### PR TITLE
add trusted algo

### PR DESCRIPTION
**Pull request by alexcos20**
## Description

Adds trusted algo

## Is this PR related with an open issue?

Related to Issue # https://github.com/oceanprotocol/ocean/issues/415

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

### CODE REVIEW

File: src/ocean/OceanCompute.ts [hunk 1]
```diff
@@ -66,16 +66,43 @@ export class OceanCompute extends Instantiable {
...
     ): SubscribablePromise<OrderProgressStep, string> {
         return new SubscribablePromise(async observer => {
             const { assets, keeper, utils } = this.ocean
             const ddo: DDO = await assets.resolve(datasetDid)
+            const service: Service = ddo.findServiceByType('compute')
+            if (!service) return null
+            if (algorithmMeta) {
+                // check if raw algo is allowed
+                if (service.attributes.main.privacy)
+                    if (!service.attributes.main.privacy.allowRawAlgorithm) return null
```
Comment (ssallam) [inline src/ocean/OceanCompute.ts:92]: It's better to have the return statement on a new line. Also why not join the if statements with an &&

File: src/ocean/OceanCompute.ts [hunk 2]
```diff
@@ -282,10 +311,13 @@ export class OceanCompute extends Instantiable {
                     datePublished,
                     price,
                     timeout: 3600,
+                    privacy: {},
                     name
                 },
                 serviceAgreementTemplate
             }
         }
+        if (computePrivacy) service.attributes.main.privacy = computePrivacy
```
Comment (ssallam) [inline src/ocean/OceanCompute.ts:320]: new line please

Comment (alexcos20) [reply src/ocean/OceanCompute.ts:320]: eslint will put it back as it is now :)

File: src/ocean/OceanCompute.ts (hunk 1 above)
Comment (alexcos20) [reply src/ocean/OceanCompute.ts:92]: /ocean/squid-js/src/ocean/OceanCompute.ts
   92:77  error  Delete `⏎························`  prettier/prettier
  321:28  error  Delete `·⏎···········`              prettier/prettier

eslint will make it again as it is now

**Pull request merged**